src/hotspot/share/opto/subnode.cpp

```diff
@@ -1928,6 +1928,38 @@ bool BoolNode::is_counted_loop_exit_test() {
   return false;
 }
 
+template<typename IntegerType>
+static const IntegerType* integral_abs_value(const IntegerType* t) {
+  typedef typename IntegerType::NativeType NativeType;
+
+  assert(!t->is_con(), "Constant types must have already been handled");
+
+  NativeType lo_abs = uabs(t->_lo);
+  NativeType hi_abs = uabs(t->_hi);
+
+  if (lo_abs < 0) {
+    assert(lo_abs == std::numeric_limits<NativeType>::min(), "uabs(t->_lo) must be min value if negative!");
```

Comment:
I think asserting `t->_lo` to be min is more straightforward, and also indicates `(t->_lo) + 1`, which yields max, is in the type. We can simplify the comment below too.